File: src/foundations_julia/vectors.jl
import Pkg 
Pkg.add("Plots")
Pkg.add("OffsetArrays")

using Plots
using OffsetArrays: Origin

function F(U,t) 
    
   x, xd = U[1], U[2]

   return  [  xd, -x ]
end 

N = 200

U =  zeros(N+1, 2)  
println( typeof(U) )

t = zeros(N+1)  
t = Vector( range(0., 4*pi, N+1) )
t = Origin(0)(t)
U = Origin(0, 1)(U) 
println(t[0])

U[0, :] = [1, 0] 
dt = (4*pi)/N 

for n in 0:N-1
   U[n+1, :] = U[n,:]  + dt * F( U[n, :], t[n] )
end

plot( U[:, 1], U[:, 2], aspect_ratio=:equal  )

File: Python/foundations_python/vectors.py
from numpy import array, zeros, linspace, sin, pi 
import matplotlib.pyplot as plt 


def F(U,t): 
    
    x, xd = U[0], U[1]

    return array( [  xd, -x ])


N = 200

U = array( zeros([N+1, 2]) ) 
t = array( zeros([N+1]) ) 
print( type(U) ) 

U[0, :] = array( [1, 0] )


dt = (4*pi)/N 
t = linspace(0., 1., N+1)

for n in range(0, N): 
  U[n+1, :] = U[n,:]  + dt * F( U[n, :], t[n] )

plt.axis("equal")
plt.plot( U[:, 0], U[:,1] )
plt.show()




